spec/solver_spec.rb

```diff
@@ -0,0 +1,55 @@
+require_relative '../lib/solver'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  context '#new' do
+    it 'returns a Solver object' do
+      expect(@solver).to be_an_instance_of(Solver)
+    end
+  end
+
+  context '#factorial' do
+    it 'takes one number as an argument and returns its factorial' do
+      expect(@solver.factorial(4)).to eq(24)
+    end
+
+    it 'takes one number as an argument and returns 1 if it is 0' do
+      expect(@solver.factorial(0)).to eq(1)
+    end
+
+    it 'takes one number as an argument and raises an exception if it is not a positive number' do
+      expect(@solver.factorial(-3)).to eq('Error: cannot apply factorial to a negative number')
```

Comment:
For information purposes, you can raise an error in the code and test for it:

Testing will look like this:
```ruby
   expect { raise StandardError, 'this message exactly'}.to raise_error('this message exactly')
```